homeassistant/components/huawei_solar/sensor.py

```diff
@@ -0,0 +1,670 @@
+        icon="mdi:home-lightning-bolt-outline",
+        native_unit_of_measurement=ELECTRIC_CURRENT_AMPERE,
+        device_class=SensorDeviceClass.CURRENT,
+        state_class=SensorStateClass.MEASUREMENT,
+    ),
+    HuaweiSolarSensorEntityDescription(
+        key=rn.STORAGE_CHARGE_DISCHARGE_POWER,
+        name="Charge/Discharge power",
+        icon="mdi:home-battery-outline",
+        native_unit_of_measurement=POWER_WATT,
+        state_class=SensorStateClass.MEASUREMENT,
+        device_class=SensorDeviceClass.POWER,
+    ),
+    HuaweiSolarSensorEntityDescription(
+        key=rn.STORAGE_TOTAL_CHARGE,
+        name="Total charge",
+        icon="mdi:battery-plus-variant",
+        native_unit_of_measurement=ENERGY_KILO_WATT_HOUR,
+        state_class=SensorStateClass.TOTAL,
+        device_class=SensorDeviceClass.ENERGY,
+    ),
+    HuaweiSolarSensorEntityDescription(
+        key=rn.STORAGE_TOTAL_DISCHARGE,
+        name="Total discharge",
+        icon="mdi:battery-minus-variant",
+        native_unit_of_measurement=ENERGY_KILO_WATT_HOUR,
+        state_class=SensorStateClass.TOTAL,
+        device_class=SensorDeviceClass.ENERGY,
+    ),
+    HuaweiSolarSensorEntityDescription(
+        key=rn.STORAGE_CURRENT_DAY_CHARGE_CAPACITY,
+        name="Day charge",
+        icon="mdi:battery-plus-variant",
+        native_unit_of_measurement=ENERGY_KILO_WATT_HOUR,
+        state_class=SensorStateClass.TOTAL_INCREASING,
+        device_class=SensorDeviceClass.ENERGY,
+    ),
+    HuaweiSolarSensorEntityDescription(
+        key=rn.STORAGE_CURRENT_DAY_DISCHARGE_CAPACITY,
+        name="Day discharge",
+        icon="mdi:battery-minus-variant",
+        native_unit_of_measurement=ENERGY_KILO_WATT_HOUR,
+        state_class=SensorStateClass.TOTAL_INCREASING,
+        device_class=SensorDeviceClass.ENERGY,
+    ),
+)
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    entry: ConfigEntry,
+    async_add_entities: AddEntitiesCallback,
+) -> None:
+    """Add Huawei Solar entry."""
+
+    update_coordinators = hass.data[DOMAIN][entry.entry_id][
+        DATA_UPDATE_COORDINATORS
+    ]  # type: list[HuaweiSolarUpdateCoordinator]
+
+    # When more than one inverter is present, then we suffix all sensors with '#1', '#2', ...
+    # The order for these suffixes is the order in which the user entered the slave-ids.
+    must_append_inverter_suffix = len(update_coordinators) > 1
+
+    entities_to_add: list[SensorEntity] = []
+    for idx, update_coordinator in enumerate(update_coordinators):
+        slave_entities: list[HuaweiSolarSensorEntity] = []
+
+        bridge = update_coordinator.bridge
+        device_infos = update_coordinator.device_infos
+
+        for entity_description in INVERTER_SENSOR_DESCRIPTIONS:
+            slave_entities.append(
+                HuaweiSolarSensorEntity(
+                    update_coordinator, entity_description, device_infos["inverter"]
+                )
+            )
+
```

Comment:
May be you can reformat a bit,e.g. the suggestion below saves 1 line. You could also consider creating a sub function to reduce the size of the large `if` statement.

```suggestion
        slave_entities += [
            HuaweiSolarSensorEntity(
                    update_coordinator, entity_description, device_infos["inverter"]
                ) for for entity_description in INVERTER_SENSOR_DESCRIPTIONS
        ]

```